### ISSUE TITLE
TypeScript importation

### ISSUE THREAD

**Issue by indieza** [femioladeji/react-slideshow#141]
Hello, when I try to deploy my web part into SharePoint Website I receive the following error:

Failed to load component "GUID" (WebPartName). Original error: loadComponent() returned an empty object for component "GUID" (WebPartName).
INNERERROR:
loadComponent() returned an empty object for component "GUID"

I take look over the documentation for TypeScript and I found there is some additional setup for the package usage.

[https://react-slideshow.herokuapp.com/typescript](Documentation)

May you give me more information about that where exactly I should create the "index.d.ts" file. Also is it a bug from the package?

**Comment by femioladeji** [comment]
@indieza does your app run locally?

The bug you reported doesn't seem to be connected with the react slideshow.

For the instruction
- Create a file named `index.d.ts` in this directory `types/react-slideshow-image`
- Open your `tsconfig.json` file and add this to the `compilerOptions` object
```
"typeRoots": [
    "./types",
    "./node_modules/@types"
  ]
```
- Then paste the content in the instruction into the file

**Comment by indieza** [comment]
The application successfully runned locally in debug mode, but when I upload the application into SharePoint a receive the error. Where I should find/create the "types" folder?

**Comment by femioladeji** [comment]
In your project root, where you have your package.json

**Comment by indieza** [comment]
![image](https://user-images.githubusercontent.com/11813571/125110726-a086ec80-e0ed-11eb-8b27-f0797e1f7e10.png)
I receive that error. Any ideas

**Comment by femioladeji** [comment]
Have you added this to the tsconfig in the compilerOptions
```
"typeRoots": [
    "./types",
    "./node_modules/@types"
  ]
 ```

**Comment by indieza** [comment]
![Uploading image.png…]()
Yes, is it correct?

**Comment by indieza** [comment]
![image](https://user-images.githubusercontent.com/11813571/125111157-3e7ab700-e0ee-11eb-8868-48344d5e82a7.png)

**Comment by femioladeji** [comment]
yeah that's correct, the `cannot find name 'function'` error should be gone now

**Comment by indieza** [comment]
it is not :( also still get the same error
![image](https://user-images.githubusercontent.com/11813571/125111629-ce206580-e0ee-11eb-8900-8e7c67738aab.png)

**Comment by femioladeji** [comment]
I don't think the `SPLoaderError.loadComponentError` is connected to the `react-slideshow` package. You can check this out for more [https://www.koskila.net/failed-to-load-component-original-error-failed-to-load-path-dependency-contosospfxwebpartlocalization-from-component-guid-contosospfxwebpart/](https://www.koskila.net/failed-to-load-component-original-error-failed-to-load-path-dependency-contosospfxwebpartlocalization-from-component-guid-contosospfxwebpart/)

**Comment by indieza** [comment]
yes, but when I uninstall the slider, everything is fine.

**Comment by femioladeji** [comment]
oooh, I'm sorry it's hard to fix this because I don't know the root cause and I can't replicate it locally.

If you see anything connected to the package in the stack trace can you please paste it here?

**Comment by indieza** [comment]
for sure there is nothing related to the slider in the stack trace. Will my some additional checks over the code and will come back if I found something.
